### PR TITLE
solution

### PR DESCRIPTION
**Pull request by illiashcherbak**

### CODE REVIEW

File: src/figuresClasses.ts [hunk 1]
```diff
@@ -1,19 +1,95 @@
...
+  public shape: Shape = 'triangle';
+
+  isInvalidTriangle(): boolean {
+    return this.a === 0
+      || this.b === 0
+      || this.c === 0
+      || this.a + this.b <= this.c
+      || this.b + this.c <= this.a
+      || this.c + this.a <= this.b;
+  }
```
Comment (oleh-yablunovskyi) [inline src/figuresClasses.ts:21]: With destructurization there will be no need to use 'this' so many times. You can try it.
**const { a, b, c } = this;**

**Review by oleh-yablunovskyi — APPROVED** [review]
Good job!

File: src/figuresClasses.ts (hunk 1 above)
Comment (illiashcherbak) [reply src/figuresClasses.ts]: Thank you!

File: src/figuresClasses.ts [hunk 2]
```diff
@@ -1,19 +1,98 @@
...
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  isInvalidTriangle(): boolean {
```
Comment (solaryasha) [inline src/figuresClasses.ts:14]: It is a cool method and I like it. Let's make it private, cause you are using it only internally

File: src/figuresClasses.ts [hunk 3]
```diff
@@ -1,19 +1,98 @@
...
+    const { a, b, c } = this;
+
+    return a === 0
+      || b === 0
+      || c === 0
+      || a + b <= c
+      || b + c <= a
+      || c + a <= b;
```
Comment (solaryasha) [inline src/figuresClasses.ts:22]: Let's store these conditions in the array and check it with the `some` method

**Review by solaryasha — CHANGES_REQUESTED** [review]

File: src/figuresClasses.ts [hunk 4]
```diff
@@ -1,19 +1,101 @@
...
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  private isInvalidTriangle(): boolean {
+    const { a, b, c } = this;
+    const checks = [
+      a === 0,
+      b === 0,
+      c === 0,
```
Comment (solaryasha) [inline src/figuresClasses.ts:19]:
```suggestion
      a <= 0,
      b <= 0,
      c <= 0,
```

**Review by solaryasha — APPROVED** [review]

**Pull request closed without merging**